### PR TITLE
Fixes address info value getter

### LINKED ISSUE #753: (possible) bug in networking model
I ssh into a running unit and run `network-get`:

image

`ops.model.NetworkInterface` however, inits looking for a `value` field which is not there; it's called `address`.

image

as a result, in a charm,
`self.model.get_binding("juju-info").network.bind_address` ends up always being `None`, which is probably not a good thing.

### PR DESCRIPTION
**Pull request by PietroPasotti**
see:
https://github.com/juju/juju/pull/14037

Fixes #753

### CODE REVIEW

File: ops/model.py [hunk 1]
```diff
@@ -655,7 +655,10 @@ class NetworkInterface:
     def __init__(self, name: str, address_info: dict):
         self.name = name
         # TODO: expose a hardware address here, see LP: #1864070.
-        address = address_info.get('value')
+
+        # 'address' is the legacy key; 'value' is to-go from juju 2.9+
```
Comment (rwcarlsen) [inline ops/model.py:659]:
```suggestion
        # 'address' is the legacy key; 'value' is the go-to from juju 2.9+
```

File: ops/model.py [hunk 2]
```diff
@@ -655,7 +655,10 @@ class NetworkInterface:
...
+        # see https://github.com/juju/juju/pull/14037
+        address = address_info.get('address') or address_info.get('value')
```
Comment (rwcarlsen) [inline ops/model.py:661]: Feels like maybe we should check/assign from the new key first (swap order here).
```suggestion
        address = address_info.get('value') or address_info.get('address')
```

**Review by rwcarlsen — APPROVED** [review]
Just a couple minor nits.

File: ops/model.py (hunk 1 above)
Comment (jameinel) [inline ops/model.py:659]:
```suggestion
        # 'address' is the legacy key; 'value' is the preferred key from juju 2.9+
```

File: ops/model.py (hunk 2 above)
Comment (jameinel) [inline ops/model.py:661]: I think you'd rather do it in the opposite order. If address is legacy it should only be read if 'value' isn't set.
```suggestion
        address = address_info.get('value') or address_info.get('address')
```
I often like the more explicit:
```suggestion
        address = address_info.get('value')
        if address is None: # Juju <2.9 only had an 'address' field
            address = address_info.get('address')
```

**Comment by PietroPasotti** [comment]
@jameinel and @rwcarlsen it looks like you agree all the way through, it's lovely to see

**Review by manadart — APPROVED** [review]
Looks good contingent upon suggestion already made.

**Pull request merged**